### PR TITLE
[SDK-2121] Add support for token validation for Organizations

### PR DESCRIPTION
**Pull request by stevehobbsdev**
This PR adds support for validating the `org_id` claim in the ID token, if the
user logs into the app in the context of an Organization.

This PR does not add documentation or an API for passing `organization` to the
authorization endpoint, but it is possible to do it as you can already pass
arbitrary parameters.

**Note:** This feature is not yet available to the general public.

## Testing

This PR adds new unit tests to cover the validation of the `org_id` claim, as
well as storing the `organization` value in the transaction, if given.

### CODE REVIEW

File: src/Auth0Client.ts [hunk 1]
```diff
@@ -482,9 +495,9 @@ export default class Auth0Client {
       throw new Error('Invalid state');
     }
 
-    if (error) {
-      this.transactionManager.remove();
+    this.transactionManager.remove();
```
Comment (stevehobbsdev) [inline src/Auth0Client.ts:498]: I simply moved this out of the `if (error)` block and removed the duplicate call.

File: src/jwt.ts [hunk 2]
```diff
@@ -176,15 +176,30 @@ export const verify = (options: JWTVerifyOptions) => {
...
+      throw new Error(
+        'Organization ID (org_id) claim must be a string present in the ID token'
+      );
+    } else if (options.organizationId !== decoded.claims.org_id) {
+      throw new Error(
+        `Organization ID (org_id) claim mismatch in the ID token; expected "${options.organizationId}", found "${decoded.claims.org_id}"`
+      );
+    }
+  }
+
```
Comment (frederikprijck) [inline src/jwt.ts:203]: Do we need to support the use-case where `options.organizationId` is not provided but `decoded.claims.org_id` exists ?

File: src/Auth0Client.ts [hunk 3]
```diff
@@ -378,7 +385,13 @@ export default class Auth0Client {
       this.worker
     );
 
-    const decodedToken = this._verifyIdToken(authResult.id_token, nonceIn);
+    const organizationId = options.organization || this.options.organization;
```
Comment (frederikprijck) [inline src/Auth0Client.ts:388]: As we are expecting `this.options.organization`, shouldn't we add `organizationId` to the `Auth0ClientOptions` ?

File: __tests__/Auth0Client.test.ts [hunk 4]
```diff
@@ -452,6 +452,29 @@ describe('Auth0Client', () => {
...
+    it('calls `tokenVerifier.verify` with the organization id given in the login method', async () => {
+      const auth0 = setup();
+      await loginWithPopup(auth0, { organization: 'test_org_123' });
+
+      expect(tokenVerifier).toHaveBeenCalledWith(
+        expect.objectContaining({
+          organizationId: 'test_org_123'
+        })
+      );
+    });
```
Comment (frederikprijck) [inline __tests__/Auth0Client.test.ts:476]: Should we test loginWithRedirect here aswell ?

File: __tests__/index.test.ts [hunk 5]
```diff
@@ -678,6 +722,32 @@ describe('Auth0', () => {
...
+        expect(tokenVerifier).toHaveBeenCalledWith({
+          id_token: TEST_ID_TOKEN,
+          nonce: TEST_RANDOM_STRING,
+          aud: 'test-client-id',
+          iss: 'https://test.auth0.com/',
+          leeway: undefined,
+          max_age: undefined,
+          organizationId: TEST_ORG_ID
+        });
+      });
```
Comment (frederikprijck) [inline __tests__/index.test.ts:749]: Do we want to use this test or move to a less mock approach ? We could combine the tests added to index.test.
ts file into a single one and avoid mocking the transactionManager.

**Review by frederikprijck — CHANGES_REQUESTED** [review]

File: src/Auth0Client.ts (hunk 3 above)
Comment (stevehobbsdev) [reply src/Auth0Client.ts:388]: We will in time - as the feature is not yet available to the public, we should not confuse developers by making this too visible in the docs or API, but that will come in a future PR.

File: __tests__/index.test.ts (hunk 5 above)
Comment (stevehobbsdev) [reply __tests__/index.test.ts:749]: Yeah I think we will do that, as we're going through a round of test refactoring anyway I figured we could do that as part of that work as I don't think this section has been moved yet (correct me if I'm wrong).

File: __tests__/Auth0Client.test.ts (hunk 4 above)
Comment (stevehobbsdev) [reply __tests__/Auth0Client.test.ts:476]: I thought I had covered everything but let me check 👍

File: __tests__/index.test.ts (hunk 5 above)
Comment (frederikprijck) [reply __tests__/index.test.ts:749]: No that is correct.

File: __tests__/Auth0Client.test.ts (hunk 4 above)
Comment (frederikprijck) [reply __tests__/Auth0Client.test.ts:476]: It is covered in index.test.ts. But I was confused why these tests ended up here and the others are in index.test.ts...

**Review by frederikprijck — APPROVED** [review]

File: __tests__/Auth0Client.test.ts (hunk 4 above)
Comment (stevehobbsdev) [reply __tests__/Auth0Client.test.ts:476]: Ah yes, to follow up on this, `loginWithRedirect` just defers all its logic to `buildAuthorizeUrl`, which has tests for this.

**Pull request merged**